### PR TITLE
Add Jest as testing framework 🃏

### PR DESCRIPTION
**Pull request by kutyel**
As promised, here is my PR to add **Jest** as testing framework for your `generator-git` 🎉 

@cpojer would be very happy 😜

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -263,6 +263,15 @@ module.exports = generators.Base.extend({
     if (this.tape && !this.mocha) testScript = 'tape'
     this.package.scripts.test += testScript
 
+    if (this.jest) {
+      delete this.package.devDependencies.nyc // nyc is not needed anymore!
```
Comment (Kikobeats) [inline app/index.js]: nice hack!

File: app/index.js [hunk 2]
```diff
@@ -263,6 +263,15 @@ module.exports = generators.Base.extend({
...
+      this.package.scripts.coveralls = 'cat ./coverage/lcov.info | coveralls'
+      this.package.scripts.test = 'jest --coverage'
+      this.package.scripts['test:w'] = 'jest --watch'
```
Comment (Kikobeats) [inline app/index.js]: better `test:watch`, what do you think?

File: app/index.js [hunk 3]
```diff
@@ -284,7 +293,7 @@ module.exports = generators.Base.extend({
       this.package.scripts.prelint = 'npm run pretty'
       this.package.scripts.pretty = 'prettier-standard index.js src/**/*.js test/**/*.js bin/**/*.js'
 
-      if (this.mocha) {
+      if (this.mocha || this.jest) {
         this.package.standard = this.fs.readJSON(this.templatePath('linter/standard.json'))
```
Comment (Kikobeats) [inline app/index.js]: hmm probably now we need to split `linter/standard.json` in two files:

- one file for mocha globals: `linters/mocha-standard-globals.json`
- one file for jest globals: `linters/jest-standard-globals.json`

File: app/index.js (hunk 2 above)
Comment (kutyel) [inline app/index.js]: Agree :+1:

File: app/index.js (hunk 3 above)
Comment (kutyel) [inline app/index.js]: Sure, incoming 😉

File: app/index.js (hunk 1 above)
Comment (kutyel) [inline app/index.js]: Thanks! 😄

**Comment by kutyel** [comment]
I managed to test it, works like a charm! 👌

File: app/index.js [hunk 4]
```diff
@@ -284,8 +293,8 @@ module.exports = generators.Base.extend({
       this.package.scripts.prelint = 'npm run pretty'
       this.package.scripts.pretty = 'prettier-standard index.js src/**/*.js test/**/*.js bin/**/*.js'
 
-      if (this.mocha) {
-        this.package.standard = this.fs.readJSON(this.templatePath('linter/standard.json'))
+      if (this.mocha || this.jest) {
+        this.package.standard = { env: [this.mocha ? 'mocha' : 'jest'] }
```
Comment (Kikobeats) [inline app/index.js]: AWESOME

Comment (kutyel) [inline app/index.js]: Thanks! I thought it was another clever hack and I added it 😉 👌

**Pull request merged**